src/main/java/frc/robot/subsystems/swerve/swervestatehelpers/SwerveStateHelper.java

```diff
@@ -0,0 +1,118 @@
+package frc.robot.subsystems.swerve.swervestatehelpers;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import frc.robot.constants.Field;
+import frc.robot.constants.MathConstants;
+import frc.robot.subsystems.swerve.Swerve;
+import frc.robot.subsystems.swerve.SwerveConstants;
+import frc.robot.subsystems.swerve.SwerveMath;
+import frc.robot.subsystems.swerve.modules.ModuleUtils;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static frc.robot.subsystems.swerve.swervestatehelpers.AimAssistUtils.getRotationAssistedSpeeds;
+
+public class SwerveStateHelper {
+
+    private final Swerve swerve;
+    private final SwerveConstants swerveConstants;
+    private final Supplier<Pose2d> robotPoseSupplier;
+    private final Supplier<Optional<Translation2d>> noteTranslationSupplier;
+
+    public SwerveStateHelper(Supplier<Pose2d> robotPoseSupplier, Supplier<Optional<Translation2d>> noteTranslationSupplier,
+            Swerve swerve) {
+        this.swerve = swerve;
+        this.swerveConstants = swerve.getConstants();
+        this.robotPoseSupplier = robotPoseSupplier;
+        this.noteTranslationSupplier = noteTranslationSupplier;
+    }
+
+    public ChassisSpeeds applyAimAssistOnSpeeds(AimAssist aimAssistState, ChassisSpeeds chassisSpeeds) {
+        return switch (aimAssistState) {
+            case SPEAKER -> getRotationAssistedSpeeds(
+                    chassisSpeeds,
+                    robotPoseSupplier.get().getRotation(),
+                    SwerveMath.getRelativeTranslation(robotPoseSupplier.get().getTranslation(), new Translation2d(0,0)).getAngle(),
+                    swerveConstants
+            );
+            case AMP -> getRotationAssistedSpeeds(
+                    chassisSpeeds,
+                    robotPoseSupplier.get().getRotation(),
+                    Field.getAngleToAmp(),
+                    swerveConstants
+            );
+            case NOTE -> handleNoteAssistState(
+                    chassisSpeeds,
+                    robotPoseSupplier,
+                    noteTranslationSupplier
+            );
+            case NONE -> chassisSpeeds;
+        };
+    }
+
+
+    private ChassisSpeeds handleNoteAssistState(
+            ChassisSpeeds speeds,
+            Supplier<Pose2d> robotPoseSupplier,
+            Supplier<Optional<Translation2d>> noteTranslationSupplier) {
+        if (noteTranslationSupplier.get().isEmpty()) {
+            return speeds;
+        }
+
+        Translation2d noteRelativeToRobot = SwerveMath.getRelativeTranslation(robotPoseSupplier.get(),
+                noteTranslationSupplier.get().get());
+
+
+        double pidGainHorizontalVelocity = swerveConstants.yMetersPIDController().calculate(
```

Comment:
i dont like the word horizontal adn dont have power to tell if its the right word here. Butt in the hole swerve and kinematics code we didnt use this word once, so try to see if you canchange to a more familiar word